File: 0x06-starwars_api/0-starwars_characters.js
#!/usr/bin/node

const axios = require('axios');

// Get Movie ID from Command Line
const movieId = process.argv[2];

if (!movieId) {
  console.log('Usage: node script.js <Movie_ID>');
  process.exit(1);
}

// Make API Request
const url = `https://swapi.dev/api/films/${movieId}/`;

axios.get(url)
  .then(response => {
    if (response.status !== 200) {
      console.error(`Error: Unable to fetch data. Status Code: ${response.status}`);
      process.exit(1);
    }

    // Extract Characters
    const characters = response.data.characters;

    // Display Characters Sequentially
    displayCharactersSequentially(characters);
  })
  .catch(error => {
    console.error(`Error: ${error.message}`);
    process.exit(1);
  });

// Display Characters Sequentially
async function displayCharactersSequentially (characters) {
  for (const characterUrl of characters) {
    try {
      const characterResponse = await axios.get(characterUrl);
      console.log(characterResponse.data.name);
    } catch (error) {
      console.error(`Error fetching character: ${error.message}`);
    }
  }
}

File: 0x06-starwars_api/0-starwars_characters.py
#!/usr/bin/python3
'''
Star War API
'''

import requests
import sys


if len(sys.argv) != 2:
    print("Usage: python script.py <Movie_ID>")
    sys.exit(1)

movie_id = sys.argv[1]

url = f"https://swapi.dev/api/films/{movie_id}/"

response = requests.get(url)

if response.status_code != 200:
    print(f"Error: Unable to fetch data. Status Code: {response.status_code}")
    sys.exit(1)

movie_data = response.json()
characters = movie_data['characters']

for character_url in characters:
    character_response = requests.get(character_url)
    character_data = character_response.json()
    print(character_data['name'])
